### PR TITLE
Quell warnings

### PR DESCRIPTION
**Pull request by notspiff**
For non-hostage code.

### CODE REVIEW

**Comment by Paxxi** [comment]
Nice!

File: xbmc/guilib/GUIFontTTF.cpp [hunk 1]
```diff
@@ -732,7 +732,7 @@ bool CGUIFontTTFBase::CacheCharacter(wchar_t letter, uint32_t style, Character *
       m_posX += -bitGlyph->left;
 
     // check we have enough room for the character
-    if ((m_posX + bitGlyph->left + bitmap.width) > static_cast<int>(m_textureWidth))
+    if ((m_posX + bitGlyph->left + bitmap.width) > m_textureWidth)
```
Comment (Rechi) [inline xbmc/guilib/GUIFontTTF.cpp]: this adds a warning (not removes one), because the left side is signed and the right side is unsigned

Comment (akva2) [inline xbmc/guilib/GUIFontTTF.cpp]: hmm. then there are platform / version  / compiler differences. this definitely quells the warning on my machine, the lhs is unsigned here.
with cast:
```
/home/akva/kode/xbmc-cmake/xbmc/guilib/GUIFontTTF.cpp: In member function ‘bool CGUIFontTTFBase::CacheCharacter(wchar_t, uint32_t, CGUIFontTTFBase::Character*)’:
/home/akva/kode/xbmc-cmake/xbmc/guilib/GUIFontTTF.cpp:735:50: warning: comparison between signed and unsigned integer expressions [-Wsign-compare]
     if ((m_posX + bitGlyph->left + bitmap.width) > static_cast<int>(m_textureWidth))
```
without cast it's silent. fun! ubuntu 16.04 / gcc 5.4 / libfreetype6-dev=2.6.1-0.1ubuntu2

Comment (akva2) [inline xbmc/guilib/GUIFontTTF.cpp]: and the result being unsigned is the correct interpretation of c(++) rules. if you add signed and unsigned numbers, the result is unsigned. and bitmap.width is unsigned. is it signed in your freetype (ftimage.h, struct FT_Bitmap_) ?

Comment (Rechi) [inline xbmc/guilib/GUIFontTTF.cpp]: I'm building with depends where freetype 2.4.7 is used. `width` is signed in this version.

Comment (akva2) [inline xbmc/guilib/GUIFontTTF.cpp]: right. i guess we need to explicitly cast both sides then to have it work everywhere.

Comment (Rechi) [inline xbmc/guilib/GUIFontTTF.cpp]: I just checked and it is unsigned since version 2.5.4. Bumping to a version equal or greater maybe another option.

Comment (akva2) [inline xbmc/guilib/GUIFontTTF.cpp]: opened https://github.com/xbmc/xbmc/pull/11422 up to others to decide what to do.

**Pull request merged**